app/validators/answers/number_answer_validator.py

```diff
@@ -71,6 +73,42 @@ def validate_value_in_limits(self):
                 self.MAXIMUM_GREATER_THAN_LIMIT, value=max_value, limit=MAX_NUMBER
             )
 
+        if isinstance(answer_source_max_value, dict):
+            source = answer_source_max_value.get("source")
+            answers_with_context = self.questionnaire_schema.numeric_answer_ranges
+            answer_source_max_value = (
+                self.questionnaire_schema.get_numeric_value_for_value_source(
+                    source, answer_source_max_value, answers_with_context
+                )
+            )
+            if (
+                isinstance(answer_source_max_value, int)
+                and answer_source_max_value > MAX_NUMBER
+            ):
+                self.add_error(
+                    self.MAXIMUM_GREATER_THAN_LIMIT,
+                    value=answer_source_max_value,
+                    limit=MAX_NUMBER,
+                )
+
+        if isinstance(answer_source_min_value, dict):
+            source = answer_source_min_value.get("source")
+            answers_with_context = self.questionnaire_schema.numeric_answer_ranges
+            answer_source_min_value = (
+                self.questionnaire_schema.get_numeric_value_for_value_source(
+                    source, answer_source_min_value, answers_with_context
+                )
+            )
+            if (
+                isinstance(answer_source_min_value, int)
+                and answer_source_min_value > MIN_NUMBER
+            ):
```

Comment:
`answer_source_min_value` is an optional dict, it wont be an int. Need to add tests to value source which would demo this not working.

This is also reassigning `answer_source_min_value`, at one point it is a value source, at line 97 it because the "referred answer" which is an optional dict, use diff variable names for clarify.